packages/std/src/msgpack.rs

```diff
@@ -0,0 +1,224 @@
+            103, 101, 163, 102, 111, 111, 166, 97, 109, 111, 117, 110, 116, 42, 164, 116, 105, 109,
+            101, 207, 255, 255, 255, 255, 255, 255, 255, 255, 165, 107, 97, 114, 109, 97, 239,
+        ];
+        (msg, serialized)
+    }
+
+    fn special_chars_test_vector() -> (SomeMsg, &'static [u8]) {
+        let msg = SomeMsg::Cowsay {
+            text: "foo\"bar\\\"bla".to_string(),
+        };
+        let serialized = &[
+            129, 166, 99, 111, 119, 115, 97, 121, 129, 164, 116, 101, 120, 116, 172, 102, 111, 111,
+            34, 98, 97, 114, 92, 34, 98, 108, 97,
+        ];
+        (msg, serialized)
+    }
+
+    #[test]
+    fn to_msgpack_vec_works() {
+        let (msg, expected) = refund_test_vector();
+        let serialized = to_msgpack_vec(&msg).unwrap();
+        assert_eq!(serialized, expected);
+
+        let (msg, expected) = release_all_test_vector();
+        let serialized = to_msgpack_vec(&msg).unwrap();
+        assert_eq!(serialized, expected);
+    }
+
+    #[test]
+    fn from_msgpack_works() {
+        let (msg, serialized) = refund_test_vector();
+        let deserialized: SomeMsg = from_msgpack(serialized).unwrap();
+        assert_eq!(deserialized, msg);
+
+        let (msg, serialized) = release_all_test_vector();
+        let deserialized: SomeMsg = from_msgpack(serialized).unwrap();
+        assert_eq!(deserialized, msg);
+    }
+
+    #[test]
+    fn from_msgpack_or_binary() {
+        let msg = SomeMsg::Refund {};
+        let serialized: Binary = to_msgpack_binary(&msg).unwrap();
+
+        let parse_binary: SomeMsg = from_msgpack(&serialized).unwrap();
+        assert_eq!(parse_binary, msg);
+
+        let parse_slice: SomeMsg = from_msgpack(serialized.as_slice()).unwrap();
+        assert_eq!(parse_slice, msg);
+    }
+
+    #[test]
+    fn from_msgpack_works_for_special_chars() {
+        let (msg, serialized) = special_chars_test_vector();
+        let deserialized: SomeMsg = from_msgpack(serialized).unwrap();
+        assert_eq!(deserialized, msg);
+    }
+
+    #[test]
+    fn deserialize_modified_field_order() {
+        #[derive(Serialize, Deserialize, Debug, PartialEq)]
+        struct TestV1 {
+            a: String,
+            b: u32,
+            c: u64,
+        }
+
+        #[derive(Serialize, Deserialize, Debug, PartialEq)]
+        struct TestV2 {
+            b: u32,
+            c: u64,
+            a: String,
+        }
+
+        let v1 = TestV1 {
+            a: "foo".to_string(),
+            b: 42,
+            c: 18446744073709551615,
+        };
+
+        let v2: TestV2 = from_msgpack(to_msgpack_vec(&v1).unwrap()).unwrap();
+        assert_eq!(
+            v2,
+            TestV2 {
+                b: 42,
+                c: 18446744073709551615,
+                a: "foo".to_string()
+            }
+        );
+    }
+
+    #[test]
+    fn deserialize_new_fields() {
+        // fields can be added, but only to the end of the struct
```

Comment:
is this a stale comment? seems this test case is adding a field in the middle of the struct...